server/src/main/java/io/crate/metadata/SimpleReference.java

```diff
@@ -221,11 +221,11 @@ public boolean isGenerated() {
     }
 
     @Override
-    public Map<String, Object> toMapping() {
+    public Map<String, Object> toMapping(int position) {
         DataType<?> innerType = ArrayType.unnest(type);
         Map<String, Object> mapping = new HashMap<>();
         mapping.put("type", DataTypes.esMappingNameFrom(innerType.id()));
-        mapping.put("position", position);
+        mapping.put("position", position >= 0 ? position : this.position);
```

Comment:
Apologies, maybe I'm confused, couldn't just use the `position`  arg and completely ignore the `this.position`? As far as I see the `position` arg is correctly calculated in `MappingUtil#addColumnProperties()`.

Reply:
This was mostly for the tests. I added a fixup to change it.

Reply:
Thx, I think it's more clear now in terms of prod code.